test/controller/pokemon.js

```diff
@@ -0,0 +1,99 @@
+const supertest = require('supertest-as-promised');
+const _ = require('lodash');
+const expect = require('chai').expect;
+describe('pokemon handling', () => {
+  let agent, otherAgent;
+  before(async () => {
+    agent = supertest.agent(sails.hooks.http.app);
+    otherAgent = supertest.agent(sails.hooks.http.app);
+    const res = await agent.post('/auth/local/register').send({
+      username: 'pk6tester',
+      password: '********',
+      email: 'pk6tester@pk6testing.com'
+    });
+    expect(res.statusCode).to.equal(302);
+    expect(res.header.location).to.equal('/');
+
+    const res2 = await otherAgent.post('/auth/local/register').send({
+      username: 'EXPLOUD_BOT',
+      password: 'AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA',
+      email: 'AAAAAAAA@AAAAAAAA.com'
+    });
+    expect(res2.statusCode).to.equal(302);
+    expect(res.header.location).to.equal('/');
+  });
+  describe('upload', () => {
+    it('should be able to upload a pk6 file and receive a parsed version', async () => {
+      const res = await agent.post('/uploadpk6').attach('pk6', __dirname + '/pkmn1.pk6');
+      expect(res.statusCode).to.equal(201);
+      expect(res.body.dexNo).to.equal(279);
+      expect(res.body.owner).to.equal('pk6tester');
+    });
+    it('should identify uploaded things as clones', async () => {
+      const res1 = await agent.post('/uploadpk6').attach('pk6', __dirname + '/pkmn2.pk6');
+      expect(res1.statusCode).to.equal(201);
+      expect(res1.body.isUnique).to.be.true;
+      const res2 = await agent.post('/uploadpk6').attach('pk6', __dirname + '/pkmn2.pk6');
+      expect(res2.statusCode).to.equal(201);
+      expect(res2.body.isUnique).to.be.false;
```

Comment:
Add a test to check that the first one is a clone after the second is uploaded. I'll merge this in first though, that can come in another PR.